microcosm_flask/conventions/swagger.py

```diff
@@ -0,0 +1,383 @@
+"""
+Swagger (OpenAPI) convention.
+
+Generates a Swagger definition, as JSON, for registered operations.
+
+Note that:
+ -  Swagger operations and type names use different conventions from the internal definitions
+    because we want to make usage friendly for code generation (e.g. bravado)
+
+ -  Marshmallow to JSON Schema conversion is somewhat simplistic. There are several projects
+    that already implement this conversion that we could try adapting. At the moment, the
+    overhead of adapting another library's conventions is too high.
+
+ -  All resource definitions are assumed to be shared and are declared in the "definitions"
+    section of the result.
+
+ -  All errors are defined generically.
+
+
+"""
+from flask import jsonify, g
+from inflection import camelize, pluralize
+from openapi import model as swagger
+from marshmallow import fields
+from werkzeug.routing import BuildError
+
+from microcosm.api import defaults
+from microcosm_flask.conventions.registry import (
+    get_qs_schema,
+    get_request_schema,
+    get_response_schema,
+    iter_operations,
+)
+from microcosm_flask.errors import ErrorSchema, ErrorContextSchema, SubErrorSchema
+from microcosm_flask.fields import EnumField
+from microcosm_flask.naming import name_for, singleton_path_for
+from microcosm_flask.operations import Operation
+
+
+def operation_name(operation, obj):
```

Comment:
The first few functions implement naming conventions for Swagger. I've chosen these to play nice with Yelp's [bravado](http://bravado.readthedocs.org/en/latest/) client.